api/src/main/java/org/openmrs/module/fhir/api/util/FHIRObsUtil.java

```diff
@@ -282,7 +283,15 @@ public static Obs generateOpenMRSObs(Observation observation, List<String> error
 				BigDecimal bd = quantity.getValue();
 				double doubleValue = bd.doubleValue();
 				obs.setValueNumeric(doubleValue);
-			}
+			}else if (FHIRConstants.ST_HL7_ABBREVATION.equalsIgnoreCase(concept.getDatatype().getHl7Abbreviation())) {
+				StringDt value = (StringDt)observation.getValue();
+				try {
+					obs.setValueAsString(value.getValue());
+				} catch (ParseException e) {
+					errors.add(" Obs set value failed");
```

Comment:
we have to log the error message. Or add errors with e. That's way we can get more info  where the error occurs.